### ISSUE TITLE
Rolling Log Files are not Respected Between runs

### ISSUE THREAD

**Issue by lucasvwamp** [log4js-node/log4js-node#842]
From the Discussion on Slack:
The rolling file appender doesn't appear to respect files between runs. If a run generates log.txt and log.1.txt.gz before terminating, the next run will start appending to log.txt, but when the file size is hit, log.1.txt.gz is appended to instead of being moved to log.2.txt.gz and a new log.1.txt.gz being created.

This can be seen by running the following file until a gz is created, stopping execution, and then starting again. Eventually, the first gz file will be appended to instead of moved. There is no difference when not using compression.

```
/* jshint esversion: 6 */
const log4js = require('log4js');
log4js.configure({
  "appenders": {
    "handler": {
      "type": "file",
      "filename": "logs/handler.log",
      "maxLogSize": 100000,
      "backups": 5,
      "keepFileExt": true,
      "compress": true
    }
  },
  "categories": {
    "default": { "appenders": ["handler"], "level": "debug" },
    "handler": { "appenders": ["handler"], "level": "debug" },
  }
});

var logsToTest = [
  "handler"
];

var logStartDate = new Date();

var loggers = logsToTest.map(log => log4js.getLogger(log));

// write out a lot
setInterval(function() {
  loggers.forEach(logger => logger.info("TESTING LOGGER!!!!!!" + logStartDate));
}, 10);
```

**Comment by nomiddlename** [comment]
I can't quite reproduce what you're describing above. The problem I see is that the file size counter resets between runs - the code doesn't check the existing size of the file and just starts counting from zero again. So if `handler.log` is 77000 bytes before you start the app, then it will not roll the file until it reaches 177000 bytes. The backup file (`handler.1.log`) will get overwritten by the contents of `handler.log` (not appended to), instead of being copied to `handler.2.log`.

**Comment by lucasvwamp** [comment]
I've also noticed what you describe. I think the appending to of the backup file only happens when compress is true.

**Comment by nomiddlename** [comment]
I can't reproduce the appending part even with compression on, but there's definitely an issue with it overwriting the backup file and not picking up the initial size.

**Comment by lucasvwamp** [comment]
I think if the overwriting backups and initial size issues are fixed, the compression appending will probably go away as well.

**Comment by nomiddlename** [comment]
Published a fix for this to NPM in version 4.1.0

**Comment by nomiddlename** [comment]
I'm going to close this issue, if the problem still exists in version 4.1.0 then please re-open.
